src/jit/nodeinfo.h

```diff
@@ -122,12 +123,21 @@ class TreeNodeInfo
     unsigned char isTgtPref:1;
     // Whether a spilled second src can be treated as a contained operand
     unsigned char regOptional:1;
+    // Whether or not a node contributes sources to its parent. Used for bookkeeping during
+    // buildRefPositionsForNode in order to avoid unnecessary work.
+    unsigned char contributesSources:1;
```

Comment:
This may seem like a minor point, but I would change "are treated as defining the transitive closure ..." to "pass through the transitive closure ...". It matters because the definition point remains at the children, not at the contained node.

---
In reply to: [72917289](https://github.com/dotnet/coreclr/pull/6533#discussion_r72917289) [](ancestors = 72917289)